### PR TITLE
Feature/user playlist filter

### PR DESCRIPTION
**Pull request by JaHajRF**
Playlist filter applied

### CODE REVIEW

File: packages/webapp/src/utils/playlistFilter.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Track } from '../store/player/player.reducer'
+
+export const playlistFilter = (searchInput: string, playlist: Track[]): Track[] => {
```
Comment (kowalski) [inline packages/webapp/src/utils/playlistFilter.ts:3]: This should be a selector. 


Than providing you have `playlist` selector that returns the full list, you can create:

```
const filteredPlaylist = (searchInput) => createSelector([playlist], (pl) => ....)
```

If you now use this in container 

```
const filteredPlaylist = useMemo(() => filteredPlaylit(searchInput), [searchInput])
const playlist = useSelector(filteredPlaylist)
```

**Review by kowalski — CHANGES_REQUESTED** [review]

File: packages/webapp/src/utils/playlistFilter.ts (hunk 1 above)
Comment (JaHajRF) [reply packages/webapp/src/utils/playlistFilter.ts:3]: That approach seems alright, but I'm using 'filterPlaylist' function as well in saga where I cannot use selector in a proper way.
As fetched songs from Audius are filtered differently than user playlist, I've applied our filter on returned songs from Audius to make filtering consistent. 
Do you suggest to change it to selector anyway? (That would duplicate some parts of the code)

Comment (kowalski) [reply packages/webapp/src/utils/playlistFilter.ts:3]: what is a problem with using selector in saga?

It seems you could do:
```
const result = yield select(filteredPlaylist)
```
right?

Comment (JaHajRF) [reply packages/webapp/src/utils/playlistFilter.ts:3]: yes. But I'm applying this filter on returned data from audius. Not redux stored playlist. So I have to pass list as an argument. 
I guess I could save that data as we receive it and create selector with the same filter for audius fetched songs. Then we would end up with two selectors -> one for playlist, second audius fetched data.

Comment (kowalski) [reply packages/webapp/src/utils/playlistFilter.ts:3]: Exactly. You should be saving result of audius search to redux. Than retrieve it using selector.

**Review by kowalski — APPROVED** [review]

**Pull request merged**